app/controllers/concerns/navigator_actions.rb

```diff
@@ -0,0 +1,156 @@
+                  (
+                    (
+                      experiments.archived = :archived OR
+                      my_modules.archived = :archived
+                    ) AND
+                    :archived IS TRUE
+                  )
+                ', archived: archived)
+                .select(
+                  'projects.id',
+                  'projects.name',
+                  'projects.archived',
+                  has_children
+                ).group('projects.id')
+  end
+
+  def load_folders(folder = nil, archived = false)
+    current_team.project_folders.where(parent_folder: folder)
+                .left_outer_joins(projects: { user_assignments: :user_role }, project_folders: {})
+                .where(project_folders: { archived: archived })
+                .where('
+                  user_assignments.user_id = ? AND
+                  user_roles.permissions @> ARRAY[?]::varchar[] OR
+                  projects.id IS NULL
+                ', current_user.id, ProjectPermissions::READ)
+                .select(
+                  'project_folders.id',
+                  'project_folders.name',
+                  'project_folders.archived',
+                  'SUM(CASE WHEN projects.id IS NOT NULL OR project_folders_project_folders.id IS NOT NULL
+                            THEN 1 ELSE 0 END) > 0 AS has_children'
+                ).group('project_folders.id')
+  end
+
+  def load_experiments(project, archived = false)
+    has_children = if !archived
+                     'SUM(CASE WHEN my_modules.archived IS FALSE THEN 1 ELSE 0 END) > 0 AS has_children'
+                   else
+                     'SUM(CASE WHEN my_modules.archived IS TRUE THEN 1 ELSE 0 END) > 0 AS has_children'
+                   end
+    project.experiments
+           .viewable_by_user(current_user, current_team)
+           .viewable_children_by_user(current_user)
+           .where('
+             experiments.archived = :archived OR
+             my_modules.archived = :archived AND
+             :archived IS TRUE
+           ', archived: archived)
+           .select(
+             'experiments.id',
+             'experiments.name',
+             'experiments.archived',
+             has_children
+           ).group('experiments.id')
+  end
+
+  def load_my_modules(experiment, archived = false)
+    experiment.my_modules
+              .viewable_by_user(current_user, current_team)
+              .where(archived: archived)
+  end
+
+  def folder_tree(folder, children, archived = false)
+    parent_folder = folder.parent_folder
+    tree = load_projects(parent_folder, archived).map { |i| project_serializer(i) } +
+           load_folders(parent_folder, archived).map { |i| folder_serializer(i) }
+    tree.find { |i| i[:id] == "PF#{folder.id}" }[:children] = children
+    tree = folder_tree(parent_folder, tree, archived) if parent_folder.present?
+    tree
+  end
+
+  def project_level(folder = nil, archived = false)
```

Comment:
these should probably be called `project_level_tree` or something, project_level is a bit too vague